pkg/storegateway/bucket.go

```diff
@@ -1859,6 +1822,20 @@ func (s *bucketBlockSet) getFor(mint, maxt int64, blockMatchers []*labels.Matche
 	return bs
 }
 
+// timerange returns the minimum and maximum timestamp available in the set.
+//
+// NOTE: s.blocks are expected to be sorted in minTime order.
+func (s *bucketBlockSet) timerange() (mint, maxt int64) {
+	if len(s.blocks) == 0 {
+		return math.MaxInt64, math.MinInt64
+	}
+
+	mint = s.blocks[0].meta.MinTime
+	maxt = s.blocks[len(s.blocks)-1].meta.MaxTime
```

Comment:
I don't think this is necessarily correct. We sort `s.blocks` by `MinTime` but there's no guarantee that the block with the highest `MinTime` also has the highest `MaxTime`. The old code didn't make this assumption AFAICT.